ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageSievingInputStream.java

```diff
@@ -79,69 +90,63 @@ public MessageSievingInputStream(InputStream stream, List<MessageInfo> messageIn
     }
 
     int bytesRead = 0;
-    byte[] data = new byte[totalMessageListSize];
-    long startTime = SystemTime.getInstance().milliseconds();
+    List<InputStream> msgStreamList = new ArrayList<>();
+    long batchStartTime = SystemTime.getInstance().milliseconds();
     logger.trace("Starting to validate message stream ");
-    int offset = 0;
     for (MessageInfo msgInfo : messageInfoList) {
       int msgSize = (int) msgInfo.getSize();
-      Utils.readBytesFromStream(stream, data, offset, msgSize);
+      // Read the entire message to create an InputStream for just this message. This is to isolate the message
+      // from the batched stream, as well as to ensure that subsequent messages can be correctly processed even if there
+      // was an error during the sieving for this message.
```

Comment:
is this an appropriate place to leave a TODO to explore `BoundedInputStream`?

after the sieve, we would just empty the stream (if message was ok, we would read nothing. If it was not, then we do what you intend here)

Reply:
Added.